picard/ui/savewarningdialog.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+#
+# Picard, the next-generation MusicBrainz tagger
+#
+# Copyright (C) 2023 Bob Swift
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+
+
+from PyQt5 import QtWidgets
+
+from picard.config import get_config
+
+from picard.ui import (
+    PicardDialog,
+    SingletonDialog,
+)
+
+
+class SaveWarningDialog(PicardDialog, SingletonDialog):
+
+    def __init__(self, parent=None):
+        super().__init__(parent)
+
+        actions = []
+        config = get_config()
+        if not config.setting["dont_write_tags"]:
+            actions.append(_("overwrite existing metadata in the files"))
+        if config.setting["rename_files"] and config.setting["move_files"]:
+            actions.append(_("rename and move the files"))
+        elif config.setting["rename_files"]:
+            actions.append(_("rename the files"))
+        elif config.setting["move_files"]:
+            actions.append(_("move the files"))
+        if actions:
+            self.WARNING_TEXT = _("This will ") + ' and '.join(actions) + '.  This action cannot be undone.  Do you want to continue?'
```

Comment:
From i18n point of view this is incorrect (this will not translate well to other languages).
That's because of the string concatenation (see https://doc.qt.io/qt-6/i18n-source-translation.html#use-parameters-instead-of-concatenating-strings)

Something like:

```
This action will:
- action1
- action2
- action3

This action cannot be undone.  Do you want to continue?
```

would work better.

The list indicator has to be in the translated string, or each list element appended to a list widget.

To keep it simple, in this case, I would use Qt rich text (pseudo html):

```python
        if not config.setting['dont_write_tags']:
            actions.append(_("overwrite existing metadata in the files"))
        if config.setting['rename_files'] and config.setting['move_files']:
            actions.append(_("rename and move the files"))
        elif config.setting['rename_files']:
            actions.append(_("rename the files"))
        elif config.setting['move_files']:
            actions.append(_("move the files"))
        if actions:
            header = _("This action will:")
            footer = _("<strong>This action cannot be undone.</strong>  Do you want to continue?")
            list_of_actions = ''
            for action in actions:
                 list_of_actions += _('<li>{action}</li>').fomat(action=action)
            self.WARNING_TEXT = _('<p>{header}</p><ul>{list_of_actions}</ul><p>{footer}</p>').format(header=header, list_of_actions=list_of_actions, footer=footer)
```

(totally untested).

Reply:
I always seem to screw up the i18n stuff.  Thanks (again) for helping me get it fixed.  I changed it based on your suggested code, but avoided translating the `'<p>{header}</p><ul>{list_of_actions}</ul><p>{footer}</p>'` part because `header`, `list_of_actions` and `footer` were already translated and the remaining html tags shouldn't be translated.  Please have a look to make sure it looks okay.